packages/next-on-pages/src/buildApplication/buildVercelOutput.ts

```diff
@@ -27,20 +22,24 @@ import {
  *
  */
 export async function buildVercelOutput(): Promise<void> {
-	const pm = await getCurrentPackageManager();
-	cliLog(`Detected Package Manager: ${pm}\n`);
+	const pm = await getPackageManager();
+	if (!pm) {
+		cliError('Error: Could not detect current package manager in use');
+		process.exit(1);
+	}
+	cliLog(`Detected Package Manager: ${pm.name} (${pm.version})\n`);
 
 	cliLog('Preparing project...');
 	await generateProjectJsonFileIfNeeded();
 
 	let tempVercelConfig: TempVercelConfigInfo | undefined;
 
-	if (pm === 'bun') {
+	if (pm.name === 'bun') {
 		// Vercel introduced proper Bun support in 32.2.1 and 32.2.4 (for monorepos), therefore we should
 		// ensure the Vercel CLI has a config file telling it to use Bun for older versions. This is done
 		// to prevent a breaking change for users who are using an older version of the Vercel CLI.
-		const vercelVersion = await getPackageVersion('vercel', pm);
-		if (vercelVersion && vercelVersion < '32.2.4') {
+		const vercelInfo = await pm.getPackageInfo('vercel');
+		if (vercelInfo && vercelInfo.version < '32.2.4') {
```

Comment:
This string comparison of semver strings is prone to problems. For example `"32.12.4" < "32.2.4"`.
Better to use a `semver` library to parse and compare.

Reply:
I totally agree, [we do use semver](https://github.com/cloudflare/next-on-pages/blob/628f6bbbfa8251b703de273f2563567c467f01b8/packages/next-on-pages/src/buildApplication/processVercelFunctions/invalidFunctions.ts#L1) and we should (have) here as well, sorry I'm not sure how we missed this 😓 , thanks a lot for noticing it 🙂

Reply:
updated 🙂